### PR TITLE
Create service to manage ForageElement state + listeners

### PR DESCRIPTION
**Pull request by devinmorgan**
# Description
[Create service to manage ForageElement state + listeners](https://github.com/teamforage/forage-android-sdk/pull/53/commits/1d902dcd9094b494e3245bfcf765e28a643db15a) 

My motivation for creating a service was to extract as much pure (business)
logic as possible to make it easy to test. Right now, a lot of business
logic is wrapped up in Android Views, and that is much harder to test

The `ElementStateManager` service is responsible for exposing a simple
API for managing the setting of event listeners for focus, blur, and
change events. It has two flavors: one for PIN and one for PANs to
reflect the subtle differences in isValid-related logic.

## Tests Added / Updated?
- YES. Added 19 unit tests for this service

## Screenshots
Screenshots are in #55

### CODE REVIEW

File: forage-android/src/test/java/com/joinforage/forage/android/core/element/state/PinElementStateManagerTest.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (djoksimo) [inline forage-android/src/test/java/com/joinforage/forage/android/core/element/state/PinElementStateManagerTest.kt:130]: rm whitespace

File: forage-android/src/main/java/com/joinforage/forage/android/core/element/ElementValidationErrors.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    detail = "Your EBT card number is invalid."
+)
+
+// PIN Input Errors
+val IncompleteEbtPinError = ElementValidationError(
+    detail = "Your EBT card PIN is incomplete."
+)
+val WrongEbtPinError = ElementValidationError(
+    detail = "Invalid EBT card PIN entered. Please enter your 4-digit PIN."
+)
```
Comment (djoksimo) [inline forage-android/src/main/java/com/joinforage/forage/android/core/element/ElementValidationErrors.kt:21]: lovin the copy 🙌🏻

**Review by djoksimo — CHANGES_REQUESTED** [review]
This is fantastic. These tests are hella clean

File: forage-android/src/test/java/com/joinforage/forage/android/core/element/state/PinElementStateManagerTest.kt (hunk 1 above)
Comment (devinmorgan) [reply forage-android/src/test/java/com/joinforage/forage/android/core/element/state/PinElementStateManagerTest.kt:130]: ooooooo shit that's my bad! good catch!

**Review by djoksimo — APPROVED** [review]
phenomenal PR @devinmorgan

File: forage-android/src/test/java/com/joinforage/forage/android/core/element/state/PinElementStateManagerTest.kt (hunk 1 above)
Comment (devinmorgan) [reply forage-android/src/test/java/com/joinforage/forage/android/core/element/state/PinElementStateManagerTest.kt]: fixed in [fb3f7e3](https://github.com/teamforage/forage-android-sdk/pull/53#pullrequestreview-1565766639)

**Review by dleis612 — APPROVED** [review]

**Review by djoksimo — APPROVED** [review]

**Pull request merged**